caddyhttp/proxy/policy.go

```diff
@@ -100,3 +104,35 @@ func (r *RoundRobin) Select(pool HostPool) *UpstreamHost {
 	}
 	return nil
 }
+
+
+// IPHash is a policy that selects hosts based on hashing the request ip
+type IPHash struct {}
+
+func hash(s string) uint32 {
+	h := fnv.New32a()
+	h.Write([]byte(s))
+	return h.Sum32()
+}
+
+
+// Select selects an up host from the pool using a round robin ordering scheme.
+func (r *IPHash) Select(pool HostPool, request *http.Request) *UpstreamHost {
+	var clientIP string
+	poolLen := uint32(len(pool))
+	clientIP, _, err := net.SplitHostPort(request.RemoteAddr);
+	if err != nil {
+		return nil
+	}
+	hash := hash(clientIP)
+	for {
+		if poolLen == 0 { break }
```

Comment:
This is definitely is not Go coding style. Please, use gofmt to format your code before commit.